### PR TITLE
Add search to current VM Tab

### PR DESCRIPTION
**Pull request by locriacyber**
Screenshot

Partially solve https://github.com/QubesOS/qubes-issues/issues/6667

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Glade backup file
+*.glade~
```
Comment (marmarta) [inline .gitignore:2]: This should not be added to git.

File: qubes_menu/__main__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from .appmenu import main
+
+if __name__ == "__main__":
+    main()
```
Comment (marmarta) [inline qubes_menu/__main__.py:4]: No newline at the end of file.

File: qubes_menu/appmenu.py [hunk 3]
```diff
@@ -264,23 +264,23 @@ def exit_app(self):
                 task.cancel()
 
 
+from .tests.mock_app import new_mock_qapp
```
Comment (marmarta) [inline qubes_menu/appmenu.py:267]: Mocking should not happen in the application code itself, but exclusively in the tests.

File: qubes_menu/desktop_file_manager.py [hunk 4]
```diff
@@ -20,11 +20,13 @@
 """
 Helper class that manages all events related to .desktop files.
 """
-import pyinotify
 import logging
 import asyncio
 import os
 import shlex
+
+from watchdog.observers import Observer
```
Comment (marmarta) [inline qubes_menu/desktop_file_manager.py:28]: Why this change? In what ways is watchdog better than pyinotify?

**Comment by marmarta** [comment]
Also, @ninavizz , could you review the UX design part here?

**Comment by ninavizz** [comment]
Oh gosh, I hadn't wanted to add Search for a while—it should be a globally-available thing, and not sit at the individual-VM level. One of the two prototypes had Search in it, and a demonstration of how Search can work when it is implemented. 

Discoverability will be a big problem if it's at the individual-Qube level, and what about things that are global—such as settings? 

Search Results are very hard, from a UX perspective, and there is so much else to get right in the appmenu, first. 

Happy to do a call Monday AM! Will ping you on Signal!

**Comment by ninavizz** [comment]
(also, I'm not able to check-out code to review it—no idea how—and, I have yet to upgrade my machine to 4.1, which I should really do this upcoming week...)

**Comment by ninavizz** [comment]
Ok, hi, I have snowboarding brain—I just now noticed this is a contributor contribution, not a Marta contribution! 

@locriacyber the plan for Search has/had been to add it in after many other features. But, we can obvs hasten that if a contributor wants to help with that, first! I commented on the issue, and offered to do a call if you'd like. I have a corporate job 9-5, but am happy to work around your needs to meet up to chat this through. TY for your interest in helping make this tool awesome!

File: qubes_menu/desktop_file_manager.py (hunk 4 above)
Comment (locriacyber) [reply qubes_menu/desktop_file_manager.py:28]: pyinotify is using `asyncore`, which is deprecated

File: qubes_menu/appmenu.py (hunk 3 above)
Comment (locriacyber) [reply qubes_menu/appmenu.py:267]: I don't want to run the app in dom0, and I need a way to run the app.

File: .gitignore (hunk 1 above)
Comment (locriacyber) [reply .gitignore:2]: Why not? `.gitignore` is used to ignore build files and temporary files.

**Comment by locriacyber** [comment]
> Also, @ninavizz , could you review the UX design part here?

There is no design thought put into it what-so-ever. I'm not good at GTK or Glade. I only know how to do full text search properly.

File: qubes_menu/appmenu.py (hunk 3 above)
Comment (marmarek) [reply qubes_menu/appmenu.py:267]: Then, create a file in `tests` to run it with `qubesadmin.Qubes()` patched. Production code should not import tests under any circumstances.

**Comment by locriacyber** [comment]
@ninavizz emailed you about meet up a week ago (to the address with qubes domain name). Maybe check your spam filter?

**Comment by ninavizz** [comment]
@locriacyber Hiya! I just started a new job and have been underwater for the past several weeks, my apologies for leaving you hanging. :( 

My primary concern, is that the Search functionality in this PR is on a per-qube basis. Preferably it would be at the parent level of the menu, and query all qubes—the way the current Search App does. 

The display of search results, and the type-ahead component, is where the UX of Search can get tricky/hard. And yes, it is also a goal/interest of mine, to see the frontend of the new app menu get to a point where it is visually "elegant" and matches the Figma prototypes. I don't code at all, so we're all depending on folks skilled at GTK to tend to that part. 

I will check my Qubes email today, tho, and get back to you asap. Sorry, again, for leaving you hanging!

**Comment by marmarek** [comment]
The search got implemented in #16
